### PR TITLE
Vinai/dolt merge p2

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
This pr brings the remaining functionality of Dolt Merge.

### CODE REVIEW

**Comment by VinaiRachakonda** [comment]
#benchmark

File: bats/sql-merge.bats [hunk 1]
```diff
@@ -70,13 +81,371 @@ SQL
     [[ "$output" =~ $head_hash ]] || false
 }
 
-@test "DOLT_MERGE with unknown branch name name throws an error" {
-    dolt sql -q "SELECT DOLT_COMMIT('-a', '-m', 'Step 1');"
+@test "DOLT_MERGE correctly merges branches with differing content in same table without conflicts" {
```
Comment (zachmu) [inline bats/sql-merge.bats:84]: Great tests

File: go/libraries/doltcore/sqle/dfunctions/dolt_merge.go [hunk 2]
```diff
@@ -158,3 +219,132 @@ func executeFFMerge(ctx *sql.Context, squash bool, dbData env.DbData, cm2 *doltd
...
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return setHeadAndWorkingSessionRoot(ctx, h)
+}
+
+func mergedRootToWorking(ctx *sql.Context, squash bool, dbData env.DbData, mergedRoot *doltdb.RootValue, cm2 *doltdb.Commit, mergeStats map[string]*merge.MergeStats) error {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dfunctions/dolt_merge.go:286]: mergeRootToWorking

**Review by zachmu — APPROVED** [review]
This looks good to me, nice work

@bheni should also take a pass before this gets checked in, as he understands the finer points of merge conflicts better than I do

**Review by bheni — COMMENTED** [review]
This PR really needs some tests of how it works when running in server mode with autocommit on and off.

**Pull request merged**